layers/+tools/lsp/packages.el

```diff
@@ -22,6 +22,15 @@
 (defun lsp/init-lsp-mode ()
   (use-package lsp-mode
     :defer t
+    :init
+    (setq lsp-cache-dir (concat spacemacs-cache-directory "lsp/")
+          lsp-server-install-dir lsp-cache-dir
+          lsp-session-file (concat lsp-cache-dir (file-name-nondirectory ".lsp-session-v1"))
+          lsp-eslint-library-choices-file (concat lsp-cache-dir ".lsp-eslint-choices")
+          lsp-yaml-schema-store-local-db (concat lsp-cache-dir "lsp-yaml-schemas.json")
+          lsp-vetur-global-snippets-dir (concat lsp-cache-dir ".snippets/vetur")
+          ;; If you find something else should be ignored, you could also set them here
```

Comment:
I don't think that setting these variables in the central LSP layer is good choice. In the end we would need to maintain one row for every layer (given that each will one day have LSP support).

I would rather prefer a solution like for the auto-complete layer. This is let the affected layers inject their individual config in a `post-init-lsp-mode` function instead. By this we keep the layer specific part localised and can still define general settings like the `lsp-cache-dir` in one place.

Reply:
Im glad to have a look at how `auto-completion` does this job.